### PR TITLE
mon: remove unused waiting_for_commit

### PR DESCRIPTION
**Pull request by tchaikov**
we are using `pending_finishers` instead of it now.

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-10-30_08:44:33-rados-wip-kefu-testing-2017-10-30-1511-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-10-31_01:59:21-rados-wip-kefu-testing-2017-10-30-1511-distro-basic-mira/

File: src/mon/Paxos.cc [hunk 1]
```diff
@@ -923,7 +923,7 @@ void Paxos::commit_finish()
   new_value.clear();
 
   // WRITING -> REFRESH
-  // among other things, this lets do_refresh() -> mon->bootstrap() know
+  // among other things, this lets do_refresh() -> mon->bootstrap() now
```
Comment (gregsfortytwo) [inline src/mon/Paxos.cc]: I don't understand this comment *either* way around off the top of my head. I think it's saying that the do_refresh() to mon->bootstrap() transition doesn't need to flush the store queue? So we're telling it that by changing states? In which case "know" is correct. But perhaps something else is happening here.

Comment (tchaikov) [reply src/mon/Paxos.cc]: i think `do_refresh() -> mon->bootstrap()` is not a transition from `do_refresh()` to `mon->bootstrap()`, it means `do_refrehs()` calls `mon->bootstrap()`.

**Comment by tchaikov** [comment]
@gregsfortytwo yes, i misunderstood it. hope it is more sensible now.

**Comment by tchaikov** [comment]
retest this please.

**Pull request merged**